python/dgl/graphbolt/dataloader.py

```diff
@@ -53,6 +54,8 @@ class MultiprocessingWrapper(dp.iter.IterDataPipe):
 
     def __init__(self, datapipe, num_workers=0, persistent_workers=True):
         self.datapipe = datapipe
+        # https://pytorch.org/docs/master/notes/multiprocessing.html#cuda-in-multiprocessing
+        mp.set_start_method("spawn", force=True)
```

Comment:
As the official [doc](https://docs.python.org/3/library/multiprocessing.html#multiprocessing.set_start_method) suggests, it's not a good idea to specify start method within `DataLoader`. And usually we call it in `__main__` like this [example](https://github.com/dmlc/dgl/blob/fda8fd224221a4ac59dc93ac22e69fd8e9d46c7f/examples/multigpu/graphbolt/node_classification.py#L411).

Reply:
I see. Let me do it inside the failing test then.

Reply:
But this is not a multi-GPU example, even though it is single GPU, it fails with multiple worker processes.

Reply:
Is my change here suitable: https://github.com/dmlc/dgl/pull/6738/commits/281da1628097d0d44c96c6159b90c9fa960c78f9